content/en/tracing/profiling/java.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Profiling Java Applications
+kind: documentation
+further_reading:
+    - link: 'https://www.datadoghq.com/blog/introducing-datadog-profiling/'
+      tags: 'Blog'
+      text: 'Introducing always-on production profiling in Datadog.'
+---
+
+The Datadog Profiler requires [Java Flight Recorder][1]. The Datadog Profiling library supports the Java JRE 11 and higher of both Oracle JDK and OpenJDK. Early-access versions of Java aren't supported. All JVM-based languages, i.e. Scala, Groovy, Kotlin, Clojure, etc. are supported.
+
+Java profiling and tracing are shipped within the same `dd-java-agent.jar` library. In order to use tracing or profiling, you must enable these functions. To enable tracing, follow the [enable tracing documentation][2].
+
+## Getting Started
+
+To begin profiling applications written in any JVM based runtime:
+
+1. Download `dd-java-agent.jar`, which contains the Java Agent class files:
+
+    ```shell
+    wget -O dd-java-agent.jar 'https://repository.sonatype.org/service/local/artifact/maven/redirect?r=central-proxy&g=com.datadoghq&a=dd-java-agent&v=LATEST'
+    ```
+
+2. Add the following JVM argument when starting your application in your IDE, Maven or Gradle application script, or `java -jar` command:
+
+    ```shell
+    -javaagent:<PATH>/dd-java-agent.jar
+    ```
+
+3. Enable profiling by adding `-Ddd.profiling.enabled=true` to the JVM command line (in addition to any existing APM Agent parameters), or use the `DD_PROFILING_ENABLED=true` environment variable.
+
+4. Because profiles are sent directly to Datadog without using the Datadog Agent, you must pass a valid [Datadog API key][3] by either:
+
+    - Adding `-Ddd.profiling.api-key.file=<PATH>`, or use the `DD_PROFILING_API_KEY_FILE=<PATH>` environment variable. This file should only contain the API key as a string.
+    - Setting the `DD_PROFILING_API_KEY=<DATADOG_API_KEY>` environment variable.
+
+5. Visualize your profiles in the [Datadog APM > Profiling page][4].
+
+**Note**: The `-javaagent` needs to be run before the `-jar` file, adding it as a JVM option, not as an application argument. For more information, see the [Oracle documentation][5]:
+
+```shell
+# Good:
+java -javaagent:dd-java-agent.jar ... -jar my-service.jar -more-flags
+# Bad:
+java -jar my-service.jar -javaagent:dd-java-agent.jar ...
+```
```

Comment:
To summarize a slack convo- some of our products differ significantly between platform/languages and should have separate pages for each. Some, like our [api](https://docs.datadoghq.com/api/), are 95% similar, so are best served with tabs for per-language differences.

Profiling is much more like the api.

You can see the ~counter-example right here.

If someone wants to get started with profiling, they land on the index. Then they'll click java and go through Getting Started above. But then the _next_ thing in a logical flow of this content is to go back to `## Profiles` on the previous page. The content below makes the most sense once they've got a profile open. So we could add links taking them back and forth between the pages as they finish relevant sections, we could copy and paste most of the content onto per-language pages, or we could have a single page with a set of tabs for Getting Started and another set of tabs for reading the header of the profiles, and then it's super easy for them to just read through the page.

Reply:
sure whatever

Reply:
Thank you! ❤️ ❤️ ❤️